docs/language_bridge.md

```diff
@@ -0,0 +1,89 @@
+Anki's codebase uses three layers.
+
+1. The web frontend, created in Svelte and typescript,
+2. The Python layer and
+3. The core Rust layer.
+
+Each layer can can makes RPC (Remote Procedure Call) to the layers below it. While it should be avoided, Python can also invoke Typescript functions. The Rust layers never make calls to the other layers. Note that it can make RPC to AnkiWeb and other servers, which is out of scope of this document.
+
+In this document we'll provide examples of bridge between languages, explaining:
+
+- where the RPC is declared,
+- where it is called (with the appropriate imports) and
+- where it is implemented.
+
+Imitating those examples should allow you to make call and create new RPCs.
+
+## Declaring RPCs
+
+Let's consider the method `NewDeck` of `DecksServices`. It's declared in [decks.proto](https://github.com/ankitects/anki/blob/acaeee91fa853e4a7a78dcddbb832d009ec3529a/proto/anki/decks.proto#L14) as `rpc NewDeck(generic.Empty) returns (Deck);`. This means this methods takes no argument (technically, an argument containing no information), and returns a [`Deck`](https://github.com/ankitects/anki/blob/acaeee91fa853e4a7a78dcddbb832d009ec3529a/proto/anki/decks.proto#L54).
+
+Read [protobuf](./protobuf.md) to learn more about how those input and output types are defined.
+
+If the RPC implementation is in Python, it should be declared in [frontend.proto](https://github.com/ankitects/anki/blob/acaeee91fa853e4a7a78dcddbb832d009ec3529a/proto/anki/frontend.proto#L24C3-L24C66)'s `FrontendService`. Otherwise, it can be declared in any other service. Obviously, the service should be consistent with what the method do.
```

Comment:
While you mention it below, you should probably explicitly state here that all RPCs are implement at the Rust layer, except when they are placed in frontend.proto.